### DISCUSSION TITLE
Calendar option

### DISCUSSION THREAD

**Discussion by debajyoti1990** [VladimirMarkelov/ttdl#73] [Ideas]
Would it be possible to make calendar option similar to taskwarrior ? In particular week number and display months in different columns like linux cal command would be excellent.

**Comment by VladimirMarkelov** [comment]
I had never used TaskWarrior, but I checked screenshots. It makes sense. I'll implement week numbers and column mode by default.  I am not sure whether there should be any settings for it. At this moment, as nobody asked anything about calendar before, I think I just "hardcode" displaying week numbers and columns,

Edit: I looked at that picture: https://taskwarrior.org/images/dark-16.png

**Comment by debajyoti1990** [comment]
Yes, those are the pictures !  I am using 1 GB WSL Debian only for taskwarrior as I needed a good calendar in terminal. And taskwarrior devs are not interested in providing linux like functionality in windows  cygwin port of taskwarrior. So I had put a discussion here. Thanks for replying.

**Comment by debajyoti1990** [comment]
Also in calendar, I think start day of the week should be defaulted to Monday, as it makes more sense to have 2 weekends after 5 days of work. :)

**Reply by VladimirMarkelov** [reply]
By the way, on that picture of TaskWarrior, Sunday is the first week day, too 😃 

While I also think that Monday should be the first week day (moreover, ISO says the same), I think I wont change anything right now. Some people might got used to the existing settings and changing the default can be a big surprise. Ideally, the application should read settings from OS, but it is also not always error-proof (e.g, in my case I use Windows with US locale, but I prefer to see Monday first). To my surprise, it appeared that more than 50% people starts a week on Sunday (from https://www.timeanddate.com/calendar/days/first-day-of-the-week.html), it is another pro for keeping things as-is.

**Comment by VladimirMarkelov** [comment]
Improved calendar in `3.8.0` release: https://github.com/VladimirMarkelov/ttdl/releases/tag/v3.8.0

**Reply by debajyoti1990** [reply]
Thanks a lot for revamping calendar. Can't thank you enough actually. I had some other questions/ideas regarding core ttdl (not calendar) . Should I start a new thread ?

**Reply by VladimirMarkelov** [reply]
If you have ideas that do not exist in TTDL, I think it make sense to open a new thread with title like "Feature requests" or "New ideas". Thank you!

**Reply by debajyoti1990** [reply]
How to print calendar such that current month is in the middle ? I did not get anything in help. For example, current month is May, how do I view calendar for March to July ?

If above is not possible , could you please add an option to view the whole year calendar in upcoming releases ?

**Reply by VladimirMarkelov** [reply]
I opened an issue #79 for watching this feature. I think, it would be the first issue I fix - because it seems the easiest one 😄

**Reply by debajyoti1990** [reply]
Thanks, I thought printing whole year calendar in 2x6 or 3x4 grid would be easier 😄.

**Reply by VladimirMarkelov** [reply]
Printing the whole year in the end of December does not look very useful 😄 For me the most interesting question is usually "what about next or two next weeks?"

**Reply by debajyoti1990** [reply]
Tbh, I was looking for something like cal command in Linux, but I think +- 3m/4m  would be good enough for me at least . 😌

**Reply by VladimirMarkelov** [reply]
Since the next release, the flag `--calendar` supports years, so you can run `ttdl --calendar=1y` and it will print the full current year

**Reply by debajyoti1990** [reply]
Thanks for supporting all the FRs, really appreciate,  after I found ttdl, it is the 2nd thing I open in my pc (1st is terminal 😄)

**Comment by debajyoti1990** [comment]
Hey, I was wondering if there is any way to view next or previous year's calendar ? Not a big use case, but good to have.

**Comment by VladimirMarkelov** [comment]
It should work but it does not. I tried and it did not show what I requested (e.g, it worked as expected for `--calendar=3m..5m` but it displayed the wrong calendar for `--calendar=-3m..1m`. I'll check what is wrong with it. Moreover, long ranges can crash it, e.g `--calendar=-2y` displays one year calendar and then crashes.

**Reply by VladimirMarkelov** [reply]
The crash is fixed. Only calendar range is left

**Reply by VladimirMarkelov** [reply]
Calendar range is fixed as well. I did not publish new version, it is fixed only in the master branch. Now you can use, e.g, `--calendar=-2y..-2y` to show 2-years old calendar.
